www/src/components/shell/onboarding/cloud/azure.tsx

```diff
@@ -0,0 +1,195 @@
+import { useCallback, useEffect, useMemo } from 'react'
+import { MenuItem, Select } from 'honorable'
+import { FormField, Input } from 'pluralsh-design-system'
+
+import { stringExists } from '../../validation'
+
+const REGIONS = [
+  'eastus',
+  'eastus2',
+  'southcentralus',
+  'westus2',
+  'westus3',
+  'australiaeast',
+  'southeastasia',
+  'northeurope',
+  'swedencentral',
+  'uksouth',
+  'westeurope',
+  'centralus',
+  'southafricanorth',
+  'centralindia',
+  'eastasia',
+  'japaneast',
+  'koreacentral',
+  'canadacentral',
+  'francecentral',
+  'germanywestcentral',
+  'norwayeast',
+  'brazilsouth',
+]
+
+export const AZURE_VALIDATIONS = [
+  {
+    field: 'credentials.workspace.region',
+    func: stringExists,
+    name: 'region',
+  },
+  {
+    field: 'credentials.workspace.project',
+    func: stringExists,
+    name: 'project',
+  },
+  {
+    field: 'credentials.azure.clientId',
+    func: stringExists,
```

Comment:
I'm pretty sure client id, tenant id, and subscription are all standard uuids, so we could validate that to prevent some misconfig

Reply:
here's a regex that matches uuids if we don't have it already in the codebase: `^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`